logging/logrus/logger.go

```diff
@@ -224,37 +229,38 @@ func (logger *Logger) GetName() string {
 
 // SetOutput sets the standard logger output.
 func (logger *Logger) SetOutput(out io.Writer) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
-	logger.std.Out = out
+	std := logger.stdStore.Load().(*lg.Logger)
+	std.Out = out
+	logger.stdStore.Store(std)
+
 }
 
 // SetFormatter sets the standard logger formatter.
 func (logger *Logger) SetFormatter(formatter lg.Formatter) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
-	logger.std.Formatter = formatter
+	std := logger.stdStore.Load().(*lg.Logger)
+	std.Formatter = formatter
```

Comment:
This is not solution since you are still updating Formatter not atommically on same instance of a logger.

I think you can find solution in standard go library - see sync/atomic/atomic_test.go TestLoadPointer(), TestStorePointer()